profiler/internal/stackparse/stackparse_test.go

```diff
@@ -0,0 +1,372 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package stackparse
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var update = flag.Bool("update", false, "update golden files")
```

Comment:
I've also implemented a [fancy version](https://godoc.org/github.com/felixge/goldy) of this [golden testing](https://about.sourcegraph.com/go/advanced-testing-in-go/#Golden-files-also-test-flags) pattern in the past. Maybe worth discussing if we agree on using this pattern in general (i'm a big fan).

Reply:
+1